Sources/_CryptoExtras/AES/Nonces.swift

```diff
@@ -0,0 +1,322 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCrypto open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftCrypto project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.md for the list of SwiftCrypto project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crypto
+import Foundation
+// MARK: - Generated file, do NOT edit
+// any edits of this file WILL be overwritten and thus discarded
+// see section `gyb` in `README` for details.
+
+fileprivate struct ByteIterator<T>: IteratorProtocol {
+    var currentOffset = 0
+    var pointer: UnsafeRawBufferPointer? = nil
+    let length: Int
```

Comment:
Great change! Let's drop the pointer though. In the fullness of time this may end up using `Span`, but for now we can still use the Array iterator in here. I don't think we want a pointer, so we'd have to use some other spelling.

Reply:
OK.  I wasn't sure if you were suggesting to also drop the offset and length. (I was trying to figure out what you meant by `Span`)

Reply:
Oh sorry, span is [here](https://forums.swift.org/t/pitch-2-safe-access-to-contiguous-storage/72745)